### PR TITLE
[PM-12279] Update SDK reference and use Origin.Android on Fido2Credential

### PR DESCRIPTION
**Pull request by aj-rosado**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-12279

## 📔 Objective

Update the SDK reference to fix EncString type downgrades
Also changed Fido2Credential in order to work with the new Origin.Android object added on https://github.com/bitwarden/sdk/pull/1042

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Review by SaintPatrck — COMMENTED** [review]
Looking good. Left a minor suggestion I think might help readability. Also, can we cover those new Error result paths with tests?

File: app/src/main/java/com/x8bit/bitwarden/data/autofill/fido2/manager/Fido2CredentialManagerImpl.kt [hunk 1]
```diff
@@ -65,11 +68,22 @@ class Fido2CredentialManagerImpl(
...
-        val origin = fido2CredentialRequest
+        val assetLinkUrl = fido2CredentialRequest
             .origin
             ?: getOriginUrlFromAttestationOptionsOrNull(fido2CredentialRequest.requestJson)
             ?: return Fido2RegisterCredentialResult.Error
 
+        val origin = Origin.Android(
+            UnverifiedAssetLink(
+                packageName = fido2CredentialRequest.packageName,
+                sha256CertFingerprint = fido2CredentialRequest.callingAppInfo
```
Comment (dseverns-livefront) [inline app/src/main/java/com/x8bit/bitwarden/data/autofill/fido2/manager/Fido2CredentialManagerImpl.kt:79]: ⛏️ with this type of chaining the style convention we've been following would put the `.callingAppInfo` on a newline too

File: app/src/main/java/com/x8bit/bitwarden/data/autofill/fido2/manager/Fido2CredentialManagerImpl.kt [hunk 2]
```diff
@@ -65,11 +68,22 @@ class Fido2CredentialManagerImpl(
...
+                    .getSignatureFingerprintAsHexString()
+                    ?: return Fido2RegisterCredentialResult.Error,
+                host = assetLinkUrl.toHostOrPathOrNull()
```
Comment (david-livefront) [inline app/src/main/java/com/x8bit/bitwarden/data/autofill/fido2/manager/Fido2CredentialManagerImpl.kt:82]: Can we get testes added for all of these error scenarios

File: app/src/test/java/com/x8bit/bitwarden/data/autofill/fido2/manager/Fido2CredentialManagerTest.kt [hunk 3]
```diff
@@ -536,6 +538,53 @@ class Fido2CredentialManagerTest {
...
+            )
+
+            val result = fido2CredentialManager.registerFido2Credential(
+                userId = "mockUserId",
+                fido2CredentialRequest = mockFido2CredentialRequest,
+                selectedCipherView = createMockCipherView(number = 1),
+            )
+
+            assertTrue(
+                result is Fido2RegisterCredentialResult.Error,
```
Comment (david-livefront) [inline app/src/test/java/com/x8bit/bitwarden/data/autofill/fido2/manager/Fido2CredentialManagerTest.kt:560]: Can we use `assertEquals` here and actually verify the contents

File: app/src/test/java/com/x8bit/bitwarden/data/autofill/fido2/manager/Fido2CredentialManagerTest.kt [hunk 4]
```diff
@@ -536,6 +538,53 @@ class Fido2CredentialManagerTest {
...
```
Comment (david-livefront) [inline app/src/test/java/com/x8bit/bitwarden/data/autofill/fido2/manager/Fido2CredentialManagerTest.kt:584]: Same thing here

File: app/src/test/java/com/x8bit/bitwarden/data/autofill/fido2/manager/Fido2CredentialManagerTest.kt (hunk 3 above)
Comment (aj-rosado) [reply app/src/test/java/com/x8bit/bitwarden/data/autofill/fido2/manager/Fido2CredentialManagerTest.kt]: changed it also on a couple other places that are similar

Comment (david-livefront) [reply app/src/test/java/com/x8bit/bitwarden/data/autofill/fido2/manager/Fido2CredentialManagerTest.kt]: Awesome!

**Review by david-livefront — APPROVED** [review]

**Pull request merged**